opencensus/metrics_quickstart/main.go

```diff
@@ -69,7 +74,14 @@ func main() {
 	if err := exporter.StartMetricsExporter(); err != nil {
 		log.Fatalf("Error starting metric exporter: %v", err)
 	}
-	defer exporter.StopMetricsExporter()
+
+	defer func() {
+		// Sleep 10 seconds (equal to ReportingInterval) to ensure updated metrics
+		// are exported at least once.
+		fmt.Println("Waiting for one more export interval to finish")
+		time.Sleep(10 * time.Second)
```

Comment:
We sleep for ~100 seconds below. Why do we need this sleep in addition? Should we all `StopMetricsExporter` after the 100 fake values?

Reply:
The problem is as follows. 
The for loop updates the metric every 1 second with different value. While exporter reads every 60 second to report back to SD backend. So after 60 seconds first report will be sent containing updates from 60 iterations.
After 100 iterations it stops metric exporter and exits. The remaining 40 updates are not reported.

Now, the problem could be fixed by reading all metrics and reporting them to the backend when the exporter is stopped. However, it needs to make sure that it waits for enough time since last report so that the backend doesn't reject the report (due to it being too soon). Typically, Stop* is not called. Exporter is started and keeps running until the life of the application. 

Hence, I added the additional sleep equal to ReportingInterval to guarantee reporting the remaining updates.

Reply:
Can we pick one, rather than doing both? Make the for loop only 60 iterations?

Or is the `defer func()` best practice that you should do every time you set up an exporter? For a production app, you probably don't want to / can't pause the application for 60 seconds before shutting it down, but you also don't want to be missing data.

Reply:
With 60 iterations it would work without having to have defer func. However, if a user changes either iteration or sleep between iteration then they will not see the correct data.
Because of the restriction on SD you cannot avoid a pause if you don't want to loose the data. Two equally bad choice, loose data or wait (rather a long one) on termination.

defer func() is not required if missing data is tolerable when the application exits.

Reply:
Maybe add a comment describing the situation, switch to 60, and don't do the defer? That way, users can add the defer if they need it, but it won't be needed for this quickstart.